ublox_derive/src/input.rs

```diff
@@ -0,0 +1,643 @@
+    let (from_fn, into_fn, rest_handling) =
+        parse_ubx_extend_attrs("#[ubx_extend_bitflags]", mac.span(), &mac.attrs)?;
+
+    let ast: BitFlagsAst = syn::parse2(mac.mac.tokens)?;
+
+    let valid_types: [(Type, u32); 3] = [
+        (syn::parse_quote!(u8), 1),
+        (syn::parse_quote!(u16), 2),
+        (syn::parse_quote!(u32), 4),
+    ];
+    let nbits = if let Some((_ty, size)) = valid_types.iter().find(|x| x.0 == ast.repr_ty) {
+        size * 8
+    } else {
+        let mut valid_type_names = String::with_capacity(200);
+        for (t, _) in &valid_types {
+            if !valid_type_names.is_empty() {
+                valid_type_names.push_str(", ");
+            }
+            valid_type_names.push_str(&t.into_token_stream().to_string());
+        }
+        return Err(Error::new(
+            ast.repr_ty.span(),
+            format!("Not supported type, expect one of {:?}", valid_type_names),
+        ));
+    };
+
+    let mut consts = Vec::with_capacity(ast.items.len());
+    for item in ast.items {
+        consts.push(BitFlagsMacroItem {
+            attrs: item.attrs,
+            name: item.name,
+            value: item.value.base10_parse()?,
+        });
+    }
+
+    Ok(BitFlagsMacro {
+        nbits,
+        vis: ast.vis,
+        attrs: ast.attrs,
+        name: ast.ident,
+        repr_ty: ast.repr_ty,
+        consts,
+        from_fn,
+        into_fn,
+        rest_handling,
+    })
+}
+
+pub fn parse_idents_list(input: proc_macro2::TokenStream) -> syn::Result<RecvPackets> {
+    syn::parse2(input)
+}
+
+fn parse_ubx_extend_attrs(
+    ubx_extend_name: &str,
+    item_sp: Span,
+    attrs: &[Attribute],
+) -> syn::Result<(
+    Option<UbxTypeFromFn>,
+    Option<UbxTypeIntoFn>,
+    Option<UbxEnumRestHandling>,
+)> {
+    let attr = attrs
+        .iter()
+        .find(|a| a.path.is_ident("ubx"))
+        .ok_or_else(|| Error::new(item_sp, format!("No ubx attribute for {}", ubx_extend_name)))?;
+    let meta = attr.parse_meta()?;
+    let mut from_fn = None;
+    let mut rest_handling = None;
+    let mut into_fn = None;
+    let meta_sp = meta.span();
+    match meta {
+        syn::Meta::List(list) => {
+            for item in list.nested {
+                if let syn::NestedMeta::Meta(syn::Meta::Path(p)) = item {
+                    if p.is_ident("from") {
+                        from_fn = Some(UbxTypeFromFn::From);
+                    } else if p.is_ident("into_raw") {
+                        into_fn = Some(UbxTypeIntoFn::Raw);
+                    } else if p.is_ident("from_unchecked") {
+                        from_fn = Some(UbxTypeFromFn::FromUnchecked);
+                    } else if p.is_ident("rest_reserved") {
+                        rest_handling = Some(UbxEnumRestHandling::Reserved);
+                    } else if p.is_ident("rest_error") {
+                        rest_handling = Some(UbxEnumRestHandling::ErrorProne);
```

Comment:
We should check that they don't specify both rest_reserved and rest_error (same for the `from_fn` fields)